spyder_terminal/widgets/main_widget.py

```diff
@@ -0,0 +1,451 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) Spyder Project Contributors
+#
+# Licensed under the terms of the MIT License
+# (see LICENSE.txt for details)
+# -----------------------------------------------------------------------------
+"""Terminal Plugin."""
+
+# Standard imports
+import os
+import sys
+import requests
+import subprocess
+import os.path as osp
+
+# Third party imports
+from qtpy.QtCore import Signal, QTimer, Slot
+from qtpy.QtWidgets import QMessageBox, QVBoxLayout
+from spyder.api.config.decorators import on_conf_change
+from spyder.api.widgets.main_widget import PluginMainWidget
+from spyder.config.base import get_translation, get_debug_level
+from spyder.utils.programs import find_program
+from spyder.widgets.tabs import Tabs
+from spyder.utils.misc import select_port
+
+# Local imports
+from spyder_terminal.api import TerminalContainerWidgetActions, TermViewMenus
+from spyder_terminal.widgets.findwidget import FindTerminal
+from spyder_terminal.widgets.terminalgui import TerminalWidget
+
+
+
+# Constants
+LOCATION = osp.realpath(osp.join(os.getcwd(),
+                                 osp.dirname(__file__)))
+WINDOWS = os.name == 'nt'
+
+# For translations
+_ = get_translation('spyder_terminal')
+
+
+class TerminalContainerWidgetToolbarSections:
+    New = 'new'
+
+
+class TerminalContainerWidgetCornerToolbar:
+    NewTerm = 'new_terminal'
+
+
+class TerminalContainerWidgetMenuSections:
+    New = 'new'
+    TabActions = 'tab_actions'
+
+
+class TerminalContainerWidget(PluginMainWidget):
```

Comment:
Please rename this class to `TerminalMainWidget`. That's because the reference to containers only apply to SpyderPluginV2's.